### ISSUE TITLE
ember-inspector Data tab is broken for new apps

### ISSUE THREAD

**Issue by Windvis** [emberjs/data#9537]
Opening an issue after discussing the issue in the Discord: https://discord.com/channels/480462759797063690/1282630945383387137

### Reproduction

1. generate a new 5.x ember app (a version that includes at least EmberData 5.3)
2. open the ember-inspector dev tools
3. click the data tab
4. A warning will be printed in the console mentioning the `@embroider/macros` package

### Description

The Data tab in the Ember inspector is broken in apps that use EmberData v5.3. The issue seems to be caused by the app reexports of the [`@ember-data/debug` data-adapter](https://github.com/emberjs/data/blob/91ca6b388c4be24b9977dc03a3f76c401aff1cab/packages/debug/package.json#L108). The reexport config points to the source file directly, which causes it to be put in the host app's "app" folder. If the host app then doesn't have `@embroider/macros` installed the macros in the `data-adapter.js` file will not be compiled away.

Installing the `@embroider/macros` in the host app is the workaround, but ideally EmberData wouldn't reexport the data-adapter like this.

Related issue: https://github.com/emberjs/ember-inspector/issues/2597
Related discussion about the app reexport config: https://discord.com/channels/480462759797063690/1282981308968210504

**Comment by runspired** [comment]
after looking into this a bit more I think this is potentially a bug in macros/embroider but I'll defer to @NullVoxPopuli @mansona or @ef4 on that.

TL;DR the DataAdapter is an app re-export. It seems that because nothing else imports it, it gets treated as an app file when it comes to evaluating the macros whereas usually re-exports get evaluated as library files. Probably a sensible optimization for a bundler to realize and hoist; but defeats our setup.

I'm not opposed to requiring `@embroider/macros` be installed as a peer, which seems like the solution here.

**Comment by ef4** [comment]
> whereas usually re-exports get evaluated as library files

Even in classic builds, addons cannot alter the transpilation of app re-exports. They're considered app code and get transpiled by whatever the app is using. That would also affect macro evaluation.

I would suggest not putting anything in treeForApp other than actual reexport statements. The v2 addon blueprint encourages this by auto-generating all the app-js reexport files.  You can keep the actual implementation in your own package's namespace.

**Comment by runspired** [comment]
@ef4 you are saying what I am saying. The package (@ember/debug) is a V2 addon. It has a file in its own code (src directory, what corresponds to the traditional addon tree) that is listed in `package.json` as an app re-export.

It appears this approach result in the library file being considered an app-file instead of a library file with a re-export in the app in the situation where there is no other library module importing the re-exported module.

**Comment by runspired** [comment]
After syncing with Ed in discord I discovered that `app-js` in the addon hash still requires a explicit re-export file, that re-export is generated by the default v2-addon setup (which we don't use since we evolved separately). Will PR a fix.
